### DISCUSSION TITLE
Is Documentation about DateTime erroneous ?

### DISCUSSION THREAD

**Discussion by Stephane-Pires** [prisma/prisma#10654] [Q&A] [answered]
## Introduction

Hi, first of all,  i'm quite new to Prisma.  

I'm actually working on a side project with the following stack : 

* NextJS
* Apollo
* Prisma
* Nexus
* Sqlite

A  lot of new library for me. 

I'm also new to GraphQL. 

------------------------

I'm trying to achieve type safety in my project. So far Prisma and nexus have helped me a lot on that part. 

But recently i have added  a GraphQL-scalar "DateTime". 

I expect to receive à Javascript Date object from Prisma. 

But it seems my Apollo-client query resolved with à String. 

Which is not what's stated in the Prisma Documentation : 
<img width="741" alt="Capture d’écran 2021-12-12 à 20 07 24" src="https://user-images.githubusercontent.com/52931491/145726133-0271108f-a9d9-4ef9-b5f5-15975373648c.png">

-------------------------

## Implementation

```js
// schema.prisma
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

// Generate the client
generator client {
  provider = "prisma-client-js"
}

// Connect to the database  (sqlite, mySQL, mongoDB, PostgreSQL)
datasource db {
  provider = "sqlite"
  url      = env("DATABASE_URL")
}

model Recipe {
  id          String   @id @unique
  name        String
  description String
  url         String
  category    String
  createdAt   DateTime @default(now())
  updatedAt   DateTime @updatedAt
}
```

```js
// /graphql/types/Recipe.ts
export const Recipe = objectType({
    name: 'Recipe',
    definition(t) {
        t.string('id')
        t.string('name')
        t.string('description')
        t.string('url')
        t.string('category')
        t.nonNull.dateTime('createdAt') // DATETIME HERE
        t.nonNull.dateTime('updatedAt')  //  DATETIME HERE
    },
})

// get Unique Recipe
export const RecipeByIDQuery = extendType({
    type: 'Query',
    definition(t) {
        t.nonNull.field('recipe', {
            type: 'Recipe',
            args: { id: nonNull(stringArg()) },
            resolve(_parent, args, ctx) {
                const recipe = ctx.prisma.recipe.findUnique({ // I expect the field "createdAt" and "updatedAt" to be resolved as Javascript Date but it seems it's resolved as "String" in the Apollo Client query
                    where: {
                        id: args.id,
                    },
                })
                return recipe
            },
        })
    },
})
```

```js
//schema.graphql
### This file was generated by Nexus Schema
### Do not make changes to this file directly


"""
A date-time string at UTC, such as 2007-12-03T10:15:30Z, compliant with the `date-time` format outlined in section 5.6 of the RFC 3339 profile of the ISO 8601 standard for representation of dates and times using the Gregorian calendar.
"""
scalar DateTime

type Query {
  recipe(id: String!): Recipe!
}

type Recipe {
  category: String
  createdAt: DateTime!
  description: String
  id: String
  name: String
  updatedAt: DateTime!
  url: String
}
```

```js
export const RecipeById = (params) => ({
    query: gql`
        query Query($recipeId: String!) {
            recipe(id: $recipeId) {
                id
                name
                description
                url
                category
                createdAt //DATETIME
                updatedAt  //DATETIME
            }
        }
    `,
    variables: { recipeId: params.id },
})
```

My VsCode has the good type  "Date" : 

<img width="856" alt="Capture d’écran 2021-12-12 à 20 22 08" src="https://user-images.githubusercontent.com/52931491/145726674-e269130b-10df-4f33-8159-2f9e4c606e0e.png">

But when i do 

```js
console.log(typeof createdAt) // string, i expect Date here
```

For me Prisma client doesn't resolve with a Javascript  Date.  Is it a bug ?

----
<img width="373" alt="Capture d’écran 2021-12-12 à 20 27 40" src="https://user-images.githubusercontent.com/52931491/145726648-f1fd96d0-38fb-4825-b57b-6febee203b80.png">

**Comment by austincrim** [comment, marked as answer]
Hey @StephanePiresLearning 👋!

This definition of the `DateTime` scalar type from the `schema.graphql` might be a clue:

> A date-time string at UTC, such as 2007-12-03T10:15:30Z, compliant with the `date-time` format outlined in section 5.6 of the RFC 3339 profile of the ISO 8601 standard for representation of dates and times using the Gregorian calendar.

This leads me to believe that Prisma returns a `Date` instance that Nexus resolves as a string which would explain the `console.log`. 

As an aside, `typeof new Date()` will return `"object"`, not `"Date"`.

**Reply by Stephane-Pires** [reply]
Yes, you are right  and also, i misunderstood the prisma doc it's stated inside the prisma doc that it returns a string

[Prisma Doc](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#default-value-for-a-datetime)

<img width="635" alt="Capture d’écran 2021-12-14 à 09 48 27" src="https://user-images.githubusercontent.com/52931491/145964696-388d68aa-745d-4c48-af37-bdddc147dcc7.png">

**Reply by janpio** [reply]
What would be a more appropriate wording here @StephanePiresLearning? A Pull Request fixing that would be very welcome.

**Reply by Stephane-Pires** [reply]
@janpio Everything is fine, the mistake is on me. 

I missconfused a javascript "Date" and a ISO 8601  "Date" 

This reply is accurate. 

> Hey @StephanePiresLearning 👋!
> 
> This definition of the `DateTime` scalar type from the `schema.graphql` might be a clue:
> 
> > A date-time string at UTC, such as 2007-12-03T10:15:30Z, compliant with the `date-time` format outlined in section 5.6 of the RFC 3339 profile of the ISO 8601 standard for representation of dates and times using the Gregorian calendar.
> 
> This leads me to believe that Prisma returns a `Date` instance that Nexus resolves as a string which would explain the `console.log`.
> 
> As an aside, `typeof new Date()` will return `"object"`, not `"Date"`.

Thanks both of you for your answers

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
